ee/hogai/graph/root/nodes.py

```diff
@@ -318,6 +323,7 @@ def run(self, state: AssistantState, config: RunnableConfig) -> PartialAssistant
 
         ui_context = self._format_ui_context(self._get_ui_context(state))
 
+        has_billing_access = self._has_billing_access(config)
```

Comment:
Question: `_has_billing_access` seems to be called twice - once here, and once in `_get_model` - is it because we can't pass this context into Temporal properly (to avoid hitting Temporal memory), or unrelated?

Reply:
I changed the logic with a slightly different function, so we only call it once.